internal/ir/xds.go

```diff
@@ -308,6 +308,9 @@ type HTTPListener struct {
 	Headers *HeaderSettings `json:"headers,omitempty" yaml:"headers,omitempty"`
 	// EnableProxyProtocol enables the listener to interpret proxy protocol header
 	EnableProxyProtocol bool `json:"enableProxyProtocol,omitempty" yaml:"enableProxyProtocol,omitempty"`
+	// ProxyProtocol provides more advanced proxy protocol configuration.
+	// If both EnableProxyProtocol and ProxyProtocol are set, ProxyProtocol takes precedence.
+	ProxyProtocol *ProxyProtocolSettings `json:"proxyProtocol,omitempty" yaml:"proxyProtocol,omitempty"`
```

Comment:
can we get rid of `EnableProxyProtocol` and rely on `ProxyProtocol{}`  instead ?